api-authorization/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,39 @@
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+const hashKey = process.env.TOKEN_SECRET;
+if (!hashKey) throw new Error('TOKEN_SECRET not found in .env');
+
+type AuthenticatedUser = {
+  userId: number;
+  username: string;
+  iat: number;
+};
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const auth = req.headers.authorization;
+  const token = auth && auth.split('Bearer ')[1];
+  if (!token) throw new ClientError(401, 'authentication required');
+  req.user = jwt.verify(token, hashKey ?? '') as AuthenticatedUser;
```

Comment:
With the check on line 6, is the `?? ''` still needed?

Reply:
yea typescript said it was still needed
![Capture](https://github.com/eric-k-chu/c-0823-code-solutions/assets/47008060/79527342-0519-4270-9476-c1d3220e5e51)